pkg/kubelet/pod_manager.go

```diff
@@ -26,68 +26,84 @@ import (
 	"github.com/golang/glog"
 )
 
+// Pod manager stores and manages access to the pods.
+//
+// Kubelet discover pod updates from 3 sources: file, http, and apiserver.
+// Pods from non-apiserver sources are called static pods, and API server is
+// not aware of the existence of static pods. In order to monitor the status of
+// such pods, kubelet create a mirror pod for each static pod via the API
+// server.
+//
+// A mirror pod has the same pod full name (name and namespace) as its static
+// counterpart (albeit different metadata such as UID, etc). By leveraging the
+// fact that kubelet reports the pod status using the pod full name, the status
+// of the mirror pod always reflects the acutal status of the static pod.
+// When a static pod gets deleted, the associated orphaned mirror pods will
+// also be removed.
+
 type podManager interface {
-	UpdatePods(u PodUpdate, podSyncTypes map[types.UID]metrics.SyncPodType)
-	GetPods() ([]api.Pod, mirrorPods)
-	GetPodByName(namespace, name string) (*api.Pod, bool)
+	GetPods() []api.Pod
 	GetPodByFullName(podFullName string) (*api.Pod, bool)
-	TranslatePodUID(uid types.UID) types.UID
-	DeleteOrphanedMirrorPods(mirrorPods *mirrorPods)
+	GetPodByName(namespace, name string) (*api.Pod, bool)
+	GetPodsAndMirrorMap() ([]api.Pod, map[string]*api.Pod)
 	SetPods(pods []api.Pod)
-	mirrorManager
+	UpdatePods(u PodUpdate, podSyncTypes map[types.UID]metrics.SyncPodType)
+	DeleteOrphanedMirrorPods()
+	TranslatePodUID(uid types.UID) types.UID
+	mirrorClient
 }
 
+// All arrays/maps in basicPodManager should be set by calling UpdatePods();
+// individual arrays/maps are not immutable and no other methods should attempt
+// to modify them.
 type basicPodManager struct {
-	// Protects all internal pod storage/mappings.
+	// Protects all internal array/maps.
 	lock sync.RWMutex
+	// Regular pods (including both static and non-static pods).
 	pods []api.Pod
-	// Record the set of mirror pods (see mirror_manager.go for more details);
-	// similar to pods, this is not immutable and is protected by the same podLock.
-	// Note that basicPodManager.pods do not contain mirror pods as they are
-	// filtered out beforehand.
-	mirrorPods mirrorPods
-
-	// A mirror pod manager which provides helper functions.
-	mirrorManager mirrorManager
-}
+	// Mirror pods created by us to monitor the static pods.
+	mirrorPods []api.Pod
```

Comment:
I *think* we don't need to keep this one, I don't see it used below. We use it in the test, but we can use one of the maps for that too.